src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java

```diff
@@ -98,14 +93,169 @@ public FileSystemItem getFileSystemItemInfo(long fsItemId, User authenticatedUse
+                                    // there is no need to remove the child entity, because it was already deleted in the recursive function call.
+                                    // if it wasn't removed (if = false) we don't remove the folder and we don't delete it.
+                                    updatedItemIds.remove(childrenEntity.getFileSystemId());
+                                    deletedEntities++;
+                                }
+                                // File
+                            } else if (childrenEntity.isFile() && fileSystemTypeRepository.findFileSystemTypeById(childrenEntity.getTypeId()) != FileSystemType.FOLDER) {
+                                // 3. Base of recursion
+                                entitiesToBeDeleted.add(childrenEntity);
+                                updatedItemIds.remove(childrenEntity.getFileSystemId());
+                                deletedEntities++;
+                            } else {
+                                throw new FileFighterDataException("FileType was wrong. " + childrenEntity);
+                            }
+                        }
+                        // otherwise the item stays as it is.
+                    } else {
+                        invisibleEntities.add(childrenEntity);
+                    }
+                }
+
+                // some entities could not be deleted because he is not allowed or cannot see them.
+                if (foundEntities.size() != deletedEntities) {
+
+                    // update itemIds.
+                    parentFileSystemEntity.setItemIds(updatedItemIds.stream().mapToLong(Long::longValue).toArray());
+
+                    // save changes of parentFolder to db.
+                    Query query = new Query().addCriteria(Criteria.where("fileSystemId").is(parentFileSystemEntity.getFileSystemId()));
+                    Update newUpdate = new Update().set("itemIds", parentFileSystemEntity.getItemIds());
+
+                    // only problem appears when there are only invisible entities left.
+                    boolean onlyInvisibleEntitiesAreLeftAfterRemovingDeletableEntities = !invisibleEntities.isEmpty() && (invisibleEntities.size() == updatedItemIds.size());
+                    if (onlyInvisibleEntitiesAreLeftAfterRemovingDeletableEntities) {
+                        // some files do not include the current user to see them. By adding up all these permissions and applying them the the parent folder
+                        // we can make sure, that the views of the other users stay the same, while the current user cannot see the folder anymore.
+                        // TODO: what are we doing if the user createdTheFolder? (we can only make it invisible if somebody else created the folder.)
+                        parentFileSystemEntity = sumUpAllPermissionsOfFileSystemEntities(parentFileSystemEntity, invisibleEntities);
```

Comment:
This Edgecase still remains to be discussed. @FileFighter/filefighter-dev

Reply:
1. make someobe else the owner?

2. empty folder remains

3. prevent this case. You can't create something in a folder of someone else and forbid them to view it.

Reply:
Really? 3. means, that users cant edit shared files (delete, rename, update) the file even though they have the permission to do it. That would me that only the owner can add files to the folder and remove them. That doesn't make much sense. I thought about 1. but that would imply that we need a runtime user.